util/dasharo/tag-with-changelog

```diff
@@ -0,0 +1,426 @@
+# a commit touched anything in the path that is checked earlier, it's counted
+# as being in that category.
+#
+# Don't run this in your current working tree, it checks out different versions
+# and can lose things.
+
+# set -x # uncomment for debug
+
+if ! { cdup="$(git rev-parse --show-cdup 2>/dev/null)" && [ -z "${cdup}" ]; }
+then
+	echo "ERROR: This is not the top directory of a git repo.  Exiting." >&2
+	exit 1
+fi
+
+# Verify the command line arguments
+if [ "$1" == "--help" ] || [ -z "$1" ] || [ -z "$2" ]; then
+	echo
+	echo "Usage: $0 <old_version> <new_version> [release notes file]"
+	echo "Old version should be a tag (4.1), a branch (origin/4.1), or a commit id"
+	echo "New version must be a name of a new tag, this script fails if it exists"
+	echo "Logfile can be omitted or it can be a new file name. In the latter case"
+	echo "a copy of the full logfile (not the final tag description) with comments"
+	echo "removed is preserved."
+	echo "Example: \"$0 origin/4.1 4.2 rnotes.txt\""
+	echo
+	echo "Note that the script starts at the commit AFTER the old version."
+	echo
+	exit 1
+else
+	OLD_GIT_VERSION="$1"
+	NEW_GIT_VERSION="$2"
+	if [[ "$OLD_GIT_VERSION" = "HEAD" || "$NEW_GIT_VERSION" = "HEAD" ]]; then
+		echo "Error: using HEAD as a reference doesn't work" >&2
+		echo
+		exit 1
+	fi
+
+	# Create a temporary lightweight tag, exit if it exists already
+	echo "Creating temporary $NEW_GIT_VERSION tag"
+	git tag $NEW_GIT_VERSION || exit 1
+
+	# Try to verify that the repo is clean before losing state.
+	if ! git diff-index --quiet --cached HEAD 2>/dev/null || \
+		[ "$(git diff $NEW_GIT_VERSION --shortstat 2>/dev/null | tail -n1)" != "" ]; then
+		echo "ERROR: repo is not clean.  Exiting." >&2
+		echo "Deleting temporary tag"
+		git tag -d $NEW_GIT_VERSION
+		exit 1
+	fi
+
+	TOTAL_COMMITS=$(git log --pretty=oneline \
+		"${OLD_GIT_VERSION}..${NEW_GIT_VERSION}" 2>/dev/null | wc -l)
+fi
+
+TOP=$(pwd)
+
+if [ -n "$3" ]; then
+	LOGFILE="${TOP}/$3"
+else
+	LOGFILE="${TOP}/relnotes.txt"
+fi
+
+# Print and log the versions
+log_versions() {
+	echo "Log of commit $1 to commit $2"
+	echo "# Log of commit $1 to commit $2" >> "$LOGFILE"
+	echo "Total commits: ${TOTAL_COMMITS}"
+	echo "# Total commits: ${TOTAL_COMMITS}" >> "$LOGFILE"
+	echo
+}
+
+# Get the first commit id in the current tree
+get_latest_commit_id() {
+(
+	cd "$1" || exit 1
+	git log -n 1 2>/dev/null | grep '^commit ' | head -1 | sed 's/commit //' | sed 's/ .*//'
+)
+}
+
+# Main get log function
+_get_log() {
+	local oldver="$1"
+	local newver="$2"
+	local title="$3"
+	local paths="$4"
+	local keywords="$5"
+
+	# Leave ${paths} unquoted in the git commands
+	# shellcheck disable=SC2086
+	{ \
+		if [ -n "$paths" ]; then \
+			git log --abbrev-commit --pretty=format:"- %s" \
+				"${oldver}..${newver}" -- ${paths} \
+				2>/dev/null; \
+		fi; \
+		if [ -n "$keywords" ]; then \
+			git log --abbrev-commit --pretty=format:"- %s" \
```

Comment:
```suggestion
			git log --abbrev-commit --pretty=tformat:"- %s" \
```
And above as well.  It makes git add newline character even after printing a single line of output, otherwise some lines get glued as you have already noticed.

Reply:
https://github.com/Dasharo/coreboot/compare/74d14b8e589dc17fe4381940f7df82be6c8ce81b..4c3f9dc7da460387649522b54bcb68cf2f6a9f31